tests/sentry/workflow_engine/handlers/condition/test_group_event_handlers.py

```diff
@@ -40,3 +46,181 @@ def test_dual_write(self):
+                    {
+                        "id": 1,
+                        "is_regression": False,
+                        "is_new": False,
+                        "is_new_group_environment": False,
+                    }
+                ),
+            }
+        )
+
+    def test_not_in_registry(self):
+        with pytest.raises(NoRegistrationExistsError):
+            attribute_registry.get("transaction")
+        dc = self.create_data_condition(
+            type=self.condition,
+            comparison=json.dumps(
+                {
+                    "match": MatchType.EQUAL,
+                    "attribute": "transaction",
+                    "value": "asdf",
+                }
+            ),
+            condition_result=True,
+        )
+        self.assert_does_not_pass(dc, self.job)
+
+    def test_equals(self):
+        dc = self.create_data_condition(
+            type=self.condition,
+            comparison=json.dumps(
+                {"match": MatchType.EQUAL, "attribute": "platform", "value": "php"}
+            ),
+            condition_result=True,
+        )
+        self.assert_passes(dc, self.job)
+
+    def test_not_equals(self):
+        dc = self.create_data_condition(
+            type=self.condition,
+            comparison=json.dumps(
+                {"match": MatchType.NOT_EQUAL, "attribute": "platform", "value": "php"}
+            ),
+            condition_result=True,
+        )
+        self.assert_does_not_pass(dc, self.job)
+
+        dc = self.create_data_condition(
+            type=self.condition,
+            comparison=json.dumps(
+                {"match": MatchType.NOT_EQUAL, "attribute": "platform", "value": "python"}
+            ),
+            condition_result=True,
+        )
+        self.assert_passes(dc, self.job)
+
+    def test_starts_with(self):
+        dc = self.create_data_condition(
+            type=self.condition,
+            comparison=json.dumps(
+                {"match": MatchType.STARTS_WITH, "attribute": "platform", "value": "ph"}
+            ),
+            condition_result=True,
+        )
+        self.assert_passes(dc, self.job)
+
+    def test_does_not_start_with(self):
+        dc = self.create_data_condition(
+            type=self.condition,
+            comparison=json.dumps(
+                {"match": MatchType.NOT_STARTS_WITH, "attribute": "platform", "value": "ph"}
+            ),
+            condition_result=True,
+        )
+        self.assert_does_not_pass(dc, self.job)
+
+        dc = self.create_data_condition(
+            type=self.condition,
+            comparison=json.dumps(
+                {"match": MatchType.NOT_STARTS_WITH, "attribute": "platform", "value": "py"}
+            ),
+            condition_result=True,
+        )
+        self.assert_passes(dc, self.job)
+
+    def test_dual_write(self):
+        comparison = {
+            "match": MatchType.EQUAL,
+            "value": "php",
+            "attribute": "platform",
+        }
```

Comment:
you can make `self.payload` on line 54 this without updating it here, you're not using it anywhere else rn